internal/overlord/servstate/handlers.go

```diff
@@ -77,17 +78,37 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 
 	_, previous := m.services[req.Name]
 	if previous {
-		return fmt.Errorf("service %q was previously started", req.Name)
+		// Already started
+		return nil
+	}
+
+	// Pass service description's environment variables to child process
+	env := os.Environ()
+	for k, v := range service.Environment {
+		env = append(env, k+"="+v)
+	}
+
+	// Expand the command string with environment variables.
+	envMap := make(map[string]string)
```

Comment:
This looks unrelated to this PR. I'm keen on supporting this ASAP as it introduces an incompatibility, but this is something worth talking about to make sure we get it right.

For example, how should "mycmd "$ASD"` expand?

I suggest taking that out of the PR so we can merge this, and let's discuss this by itself.

Reply:
This is in #58 so I'll remove it from there and propose a separate PR we can drill into this more with.